src/test/searchAddressFromZipcode.test.ts

```diff
@@ -0,0 +1,105 @@
+import { expect } from '@jest/globals';
+import { addresses } from './constants/masterData/address';
+import { expectedadAddressDataListIncludHyphen } from './constants/masterData/address';
+import { expectedadAddressDataListExcludingHyphen } from './constants/masterData/address';
+import { fullinputZipcodeExcludingHyphen } from './constants/userInput/zipcode';
+
+test('ユーザー入力から取得した郵便番号の全角数字を半角に変換 ハイフンが入っていても数字のみの抽出', () => {
+  const testDataExcludingHyphen = '７３２００２１';
+  const testDataIncludingHyphen = '７３２ー００２１';
+  const testDatamixtureFullAndHalf = '７３2００２1';
+  const expectedDataExcludingHyphen = '7320021';
+
+  const convertZipCode = (testData: string): string => {
+    const a: string = testData.replace(/[０-９]/g, (s: string) =>
+      String.fromCharCode(s.charCodeAt(0) - 65248)
+    );
+    const b: RegExpMatchArray | [] = a.match(/\d/g) || [];
+    return b.join('');
+  };
+
+  expect(convertZipCode(testDataExcludingHyphen)).toBe(
+    expectedDataExcludingHyphen
+  );
+  expect(convertZipCode(testDataIncludingHyphen)).toBe(
+    expectedDataExcludingHyphen
+  );
+  expect(convertZipCode(testDatamixtureFullAndHalf)).toBe(
+    expectedDataExcludingHyphen
+  );
+});
+
+test('検索条件と郵便番号の桁数によって処理するか否かのフラグを返す', () => {
+  const isExactTrue = true;
+  const isExactFalse = false;
+  const lengthNotSeven = 8;
+  const lengthSeven = 7;
+  const lengthLessThanTwo = 1;
+  const lengthTwo = 2;
+  let flag = true;
+
+  const checkLength = (isExact: boolean, length: number) => {
+    // 完全一致検索の場合は入力データが7文字ちょうどで処理を実行
+    if (isExact === true && length !== 7) {
+      return (flag = false);
+    }
+
+    // 部分一致検索の場合は入力データが2文字以上で処理を実行
```

Comment:
@marumoto-goq 
以下がわかりやすい。
```suggestion
    // 部分一致検索かつ入力データが2文字より少ない場合
```

Reply:
@iwata-goq 
なるほど！わかりやすくなりました！ありがとうございます！
ふと思ったのですが、そうなるとコードと書いてあることが全く同じになるので、そもそもコメントはしなくてもいいのかなと思ったのですが、ご意見聞きたいです🙇‍♂️

Reply:
@marumoto-goq 
それでもいいです！

実際に使用されている部分でコメントしていますし、問題ないかと。

Reply:
@iwata-goq 
ありがとうございます！
削除しておきます！